examples/basic_addition.exs

```diff
@@ -2,41 +2,30 @@
 
 # Initialize the local client, this just runs with default options on CPU
 client = Exla.Client.create_client()
+builder = Exla.Builder.new("addition")
 
 # We'll start by declaring the shapes of our parameters
-arg1_shape = Exla.NIF.make_scalar_shape(:int32)
-arg2_shape = Exla.NIF.make_scalar_shape(:int32)
+arg1_shape = Exla.Shape.make_shape(:int32, {})
+arg2_shape = Exla.Shape.make_shape(:int32, {})
 
 # Now we declare the parameters and a constant
-x = Exla.NIF.parameter(0, arg1_shape, "x")
-y = Exla.NIF.parameter(1, arg2_shape, "y")
-c = Exla.NIF.constant_r0(1)
+x = Exla.Op.parameter(builder, 0, arg1_shape, "x")
+y = Exla.Op.parameter(builder, 1, arg2_shape, "y")
+c = Exla.Op.constant(builder, 1)
 
 # Now we get the result, this is the root op
-z = Exla.NIF.add(Exla.NIF.add(x, y), c)
+z = Exla.Op.add(Exla.Op.add(x, y), c)
 
 # Build computation from root
-comp = Exla.NIF.build(z)
+comp = Exla.Builder.build(z)
 
 # Compile to a local executable, we must include argument shapes so it can be built in to the executable
-exec = Exla.NIF.compile(comp, {arg1_shape, arg2_shape}, %{})
+exec = Exla.Client.compile(client, comp, {arg1_shape, arg2_shape})
```

Comment:
This is interesting... why do we need to pass the arg1_shape and arg2_shape both when defining the computations parameters and when compiling? Shouldn't XLA be inferring based on the parameters that those are our shapes? 

Btw, parameters list in Elixir is typically done as a list (such as `apply(Kernel :+ [1, 2, 3])`), so should they be a list here? 🤔

Reply:
Another question, when we compile, does XLA tell us what is going to be outgoing shape? Or just when we execute it?

Reply:
It doesn't tell us the shape, but OneFlow does some things to save memory by reusing buffers with the output, so I was planning on doing most of the same.